cpp/src/lance/io/reader_test.cc

```diff
@@ -12,7 +12,7 @@
 //  See the License for the specific language governing permissions and
 //  limitations under the License.
 
-#include "lance/arrow/reader.h"
+#include "lance/io/reader.h"
```

Comment:
why move this?

Reply:
I removed the public FileReader interface. We did not use it publicly before, and it only acts a proxy to `lance::io::FileReader`. 

So this PR needs to pass Schema / Manifest to FileReader, if we are still using `lance::arrow::FileReader` it means that we need to public `Schema / Manifest` as well. I feel that it is not necessary in the moment.